lints/mozilla/lint_mp_modulus_must_be_2048_bits_or_more.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/********************************************************************
+Section 5.1 - Algorithms
+RSA keys whose modulus size in bits is divisible by 8, and is at least 2048.
+********************************************************************/
+
+package lints
+
+import (
+	"crypto/rsa"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/lint"
+	"github.com/zmap/zlint/util"
+)
+
+type modulus2048OrMore struct{}
+
+func (l *modulus2048OrMore) Initialize() error {
+	return nil
+}
+
+func (l *modulus2048OrMore) CheckApplies(c *x509.Certificate) bool {
+	return c.PublicKeyAlgorithm == x509.RSA
+}
+
+func (l *modulus2048OrMore) Execute(c *x509.Certificate) *lint.LintResult {
+	pubKey, ok := c.PublicKey.(*rsa.PublicKey)
+	if !ok {
+		return &lint.LintResult{
+			Status:  lint.Fatal,
+			Details: "certificate public key was not an RSA public key",
+		}
+	}
+
+	if bitLen := pubKey.N.BitLen(); bitLen < 2048 {
+		return &lint.LintResult{Status: lint.Error}
```

Comment:
Will this incorrectly trigger if there's a leading zero in the key? Can we add a test for this?

Reply:
Good question. I'll look into it.

Reply:
(_Bracing myself to be proven wrong by the ASN.1/PKI masters :grimacing:_)

I believe using a modulus with a leading 0 will fail to unmarshal before the lint is ever run. The stdlib `asn1` encoding package enforces that integers are minimally encoded before unmarshaling the big int into the RSA public key struct: https://github.com/golang/go/blob/master/src/encoding/asn1/asn1.go#L76-L89

I think because of that it will also be difficult to write a meaningful unit test.

As one data-point this is the same approach we've been using in Boulder for a long time as part of our `goodkey` package: https://github.com/letsencrypt/boulder/blob/925540d7bebd3145bca3f6469a06787b5078f0ea/goodkey/good_key.go#L225-L240

Reply:
Also, can confirm that `math/big` only counts bits to the right of the first 1 bit: https://play.golang.org/p/ojPDbPcUUK9